ansible_rulebook/job_template_runner.py

```diff
@@ -0,0 +1,232 @@
+    async def run_job_template(
+        self,
+        name: str,
+        organization: str,
+        job_params: dict,
+        event_handler: Callable[[dict], Any],
+    ) -> str:
+        job_template = await self.get_job_template(name, organization)
+        job = await self.launch(job_template["id"], job_params)
+
+        url_info = urlparse(job["url"])
+        url = f"{url_info.path}/job_events/"
+        counters = []
+        params = dict(parse_qsl(url_info.query))
+
+        async with aiohttp.ClientSession(
+            headers=self._auth_headers()
+        ) as session:
+            while True:
+                # fetch and process job events
+                response = await self._get_page(session, url, params)
+                json_body = json.loads(response["body"])
+                job_status = None
+                for event in json_body["results"]:
+                    job_status = dpath.get(
+                        event, "summary_fields.job.status", "."
+                    )
+                    counter = event["counter"]
+                    if counter not in counters:
+                        counters.append(counter)
+                        print(event["stdout"])
+                    if event_handler:
+                        await event_handler(event)
+
+                if json_body.get("next", None):
+                    params["page"] = params.get("page", 1) + 1
+                    continue
+
+                if job_status in self.JOB_COMPLETION_STATUSES:
+                    # fetch and return job object containing artifacts
+                    response = await self._get_page(session, url_info.path, {})
+                    return json.loads(response["body"])
+                    break
+                await asyncio.sleep(self.refresh_delay)
+
+    async def launch(self, job_template_id: int, job_params: dict) -> dict:
+        url = urljoin(
+            self.host, f"{self.JOB_TEMPLATE_SLUG}/{job_template_id}/launch/"
+        )
+
+        async with aiohttp.ClientSession(
+            headers=self._auth_headers()
+        ) as session:
+            async with session.post(url, json=job_params) as post_response:
+                response = dict(
+                    status=post_response.status,
+                    body=await post_response.text(),
+                )
+
+                if response["status"] not in self.VALID_POST_CODES:
+                    raise ControllerApiException(
+                        "Failed to post to %s. Status: %s, Body: %s"
+                        % (
+                            url,
+                            response["status"],
+                            response.get("body", "empty"),
+                        )
+                    )
+                json_body = json.loads(response["body"])
+        return json_body
+
+
+job_template_runner = JobTemplateRunner("<controller host>", "<api token>")
+
+
+# For testing only. To be removed after intergration with rulebook action
```

Comment:
@bzwei is this going to get removed for this PR?

Reply:
yes. will be removed in another PR when CLI option is added.